src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestState.cs

```diff
@@ -16,6 +16,15 @@ namespace System.Net.Http
 {
     internal sealed class WinHttpRequestState : IDisposable
     {
+#if DEBUG
+        private static int s_dbg_allocated = 0;
+        private static int s_dbg_pin = 0;
```

Comment:
Should there be a Debug.Assert on these getting back to 0 at some point to detect other kinds of leaks?